Kernel/Devices/Loop/LoopDevice.cpp

```diff
@@ -20,10 +20,18 @@ SpinlockProtected<LoopDevice::List, LockRank::None>& LoopDevice::all_instances()
     return s_all_instances;
 }
 
-void LoopDevice::remove(Badge<DeviceControlDevice>)
+ErrorOr<void> LoopDevice::remove(Badge<DeviceControlDevice>)
 {
-    LoopDevice::all_instances().with([&](auto&) {
+    // NOTE: It's safe to rely on the all_instances spinlock as the only
```

Comment:
Re-thinking this statement, it might be possible to get an instance by manually making a device node and trying to open it. So we should have a safer method to handle this.